yangvalidator/yangvalidator/v2/views.py

```diff
@@ -337,90 +309,44 @@ def create_output(request, yang_models: str, url, latest: bool, working_dir: str
-        }
+        json_body['dependencies'] = {'missing': missing, 'existing': existing_dependencies}
         if len(missing) == 0 and not found_repo_modules:
-            http_response = validate(request, xym_response, json_body)
-            if os.path.exists(working_dir) and remove_working_dir:
-                shutil.rmtree(working_dir)
-            return http_response
-        return HttpResponse(json.dumps({'output': json_body}, cls=DjangoJSONEncoder),
-                            status=202, content_type='application/json')
+            response = validate(request, xym_response, json_body)
+        else:
+            response = JsonResponse({'output': json_body}, status=202)
     elif latest:
-        json_body = {
-            'modules-to-validate': {
-                'user-modules': extracted_modules
-            },
-            'dependencies': {
-                'repo-modules': checker.get_latest_revision()
-            },
-            'cache': working_dir.split('/')[-1]
-        }
-
-        http_response = validate(request, xym_response, json_body)
-        if os.path.exists(working_dir) and remove_working_dir:
-            shutil.rmtree(working_dir)
-        return http_response
+        json_body['dependencies'] = {'repo-modules': checker.get_latest_revision()}
+        response = validate(request, xym_response, json_body)
     elif len(missing) == 0:
-        json_body = {
-            'modules-to-validate': {
-                'user-modules': extracted_modules
-            },
-            'cache': working_dir.split('/')[-1]
-        }
-        http_response = validate(request, xym_response, json_body)
-        return http_response
+        response = validate(request, xym_response, json_body)
     else:
+        json_body['dependencies'] = {'missing': missing}
         if xym_response is not None:
-            response =\
-                {
-                    'modules-to-validate': {
-                        'user-modules': extracted_modules
-                    },
-                    'xym': xym_response,
-                    'dependencies': {
-                        'missing': missing,
-                    },
-                    'cache': working_dir.split('/')[-1]
-                }
-        else:
-            response =\
-                {
-                    'modules-to-validate': {
-                        'user-modules': extracted_modules
-                    },
-                    'dependencies': {
-                        'missing': missing,
-                    },
-                        'cache': working_dir.split('/')[-1]
-                }
-        return HttpResponse(json.dumps({'output': response}, cls=DjangoJSONEncoder),
-                            status=202, content_type='application/json')
+            json_body['xym'] = xym_response
+        response = JsonResponse({'output': json_body}, status=202)
+    if os.path.exists(working_dir) and remove_working_dir:
+        shutil.rmtree(working_dir)
```

Comment:
You are removing working dir here after _yangvalidator/v2/rfc_ or _yangvalidator/v2/draft_ request, but it might be is still needed in following _yangvalidator/v2/validate_ request.
It will raise exception during shutil.copy on line 106